### PR TITLE
Movie Project - Linda Isell

### PR DESCRIPTION
**Pull request by liissee**
Link to my deployed project:
https://movie-project-linda.netlify.com/

**My reflections on how this week's project turned out:**
Really fun project, again. I started early and the videos were very helpful. I think it's hard and I have not made friends with props yet.. 

**Things I'd like to have clarified or explained in more detail:**
 - I managed to "loop" the pages so that when you click on an actor you see all the movies she/he was in and then you can click to see that movie details but I'm not sure how I did that! Can you please explain why it suddenly worked? 
- If you click on an actor it will say "Movies with Marlon Brando". I want it to display the name of the actors you clicked on but I can figure out how to do that? In this solution, I have fetched another api-link but I don't know how to add the "credit_id" in the URL, (that's why it says Marlon Brando on every page, I took his credit_id)?
Or I want to pass {casts.name} as a props (I guess?) from "MovieDetails.js" to display in "CastDetails".. Is this possible?

### CODE REVIEW

File: code/src/components/dropdownlist.css [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+.select-css option {
+  font-weight:normal;
+}
+
+/* Support for rtl text, explicit support for Arabic and Hebrew.*/
+*[dir="rtl"] .select-css, :root:lang(ar) .select-css, :root:lang(iw) .select-css {
+  background-position: left .7em top 50%, 0 0;
+  padding: .6em .8em .5em 1.4em;
+}
```
Comment (ericaglimsholt) [inline code/src/components/dropdownlist.css]: Wow!

File: code/src/components/dropdownlist.css [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+/* Disabled styles.. What is this doing? */
```
Comment (ericaglimsholt) [inline code/src/components/dropdownlist.css]: Disabled is for example a button can have `disabled` to it (when you can't submit a form for example)

File: code/src/pages/CastDetails.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  const api_key = "363444609247127238629594b245e069"
+
+  //https://api.themoviedb.org/3/person/192/52fe4210c3a36847f800135f?api_key=363444609247127238629594b245e069&language=en-US
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/person/${castId}/movie_credits?api_key=${api_key}&language=en-US`)
+      .then((res) => res.json())
+      .then((json) => {
+        setPerson(json.cast)
+        console.log(json.cast)
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]:
```suggestion
```

File: code/src/pages/CastDetails.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+export const CastDetails = () => {
+  const { castId } = useParams()
+
+  const [person, setPerson] = useState([])
+  const [actorName, setActorName] = useState("")
+
+  const api_key = "363444609247127238629594b245e069"
+
+  //https://api.themoviedb.org/3/person/192/52fe4210c3a36847f800135f?api_key=363444609247127238629594b245e069&language=en-US
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]:
```suggestion
```

File: code/src/pages/CastDetails.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    fetch(`https://api.themoviedb.org/3/person/${castId}/movie_credits?api_key=${api_key}&language=en-US`)
+      .then((res) => res.json())
+      .then((json) => {
+        setPerson(json.cast)
+        console.log(json.cast)
+      })
+  }, [castId])
+
+
+  //https://api.themoviedb.org/3/credit/52fe4210c3a36847f800135f?api_key=363444609247127238629594b245e069
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]:
```suggestion
```

File: code/src/pages/CastDetails.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/credit/52fe4210c3a36847f800135f?api_key=${api_key}`)
+      .then((res) => res.json())
+      .then((json) => {
+        setActorName(json.person)
+        console.log(json.person)
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]:
```suggestion
```

File: code/src/pages/CastDetails.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            </div>
+          </Link>
+        ))}
+      </section>
+    </div>
+  )
+}
+
+// I managed to "loop" the pages so that when you click on a actor you see all the 
+// movies she/he was in and then you can click to see that movie details but I'm not sure how I did that!
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]: Take a look at your url! That's how :D

File: code/src/pages/MovieDetails.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+
+export const MovieDetails = () => {
+  const [movieDetails, setMovieDetails] = useState([])
+  const { id } = useParams()
+  const [loading, setLoading] = useState(false)
+  const [cast, setCast] = useState([])
+
+  const api_key = "363444609247127238629594b245e069"
+  // https://api.themoviedb.org/3/movie/425?api_key=363444609247127238629594b245e069&language=en-US
```
Comment (ericaglimsholt) [inline code/src/pages/MovieDetails.js]:
```suggestion
```

File: code/src/pages/MovieDetails.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+  useEffect(() => {
+    setLoading(true)
+    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=${api_key}&language=en-US`)
+      .then((res) => res.json())
+      .then((json) => {
+        console.log(json)
```
Comment (ericaglimsholt) [inline code/src/pages/MovieDetails.js]:
```suggestion
```

File: code/src/pages/MovieDetails.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        setMovieDetails(json)
+        setLoading(false)
+      })
+  }, [id])
+
+  //https://api.themoviedb.org/3/movie/330457/credits?api_key=363444609247127238629594b245e069  test link.
```
Comment (ericaglimsholt) [inline code/src/pages/MovieDetails.js]:
```suggestion
```

File: code/src/pages/MovieDetails.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${id}/credits?api_key=${api_key}`)
+      .then((res) => res.json())
+      .then((json) => {
+        setCast(json.cast.slice(0, 3))
+        console.log(json.cast.slice(0, 3))
```
Comment (ericaglimsholt) [inline code/src/pages/MovieDetails.js]:
```suggestion
```

File: code/src/pages/MovieList.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  const [movies, setMovies] = useState([])
+  const [chosenCategory, setChosenCategory] = useState("popular")
+
+  const api_key = "363444609247127238629594b245e069"
+
+  const chosenMovieList = category => {
+    setChosenCategory(category)
+  }
+
+  // https://api.themoviedb.org/3/movie/popular?api_key=363444609247127238629594b245e069&language=en-US&page=1
```
Comment (ericaglimsholt) [inline code/src/pages/MovieList.js]:
```suggestion
```

File: code/src/pages/MovieList.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${chosenCategory}?api_key=${api_key}&language=en-US&page=1`)
+      .then((res) => res.json())
+      .then((json) => {
+        setMovies(json.results)
+        console.log(json)
```
Comment (ericaglimsholt) [inline code/src/pages/MovieList.js]:
```suggestion
```

File: code/src/pages/CastDetails.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      .then((res) => res.json())
+      .then((json) => {
+        setActorName(json.person)
+        console.log(json.person)
+      })
+  }, [])
+
+  return (
+    <div className="cast-details-top">
+      <h1>Movies with {actorName.name}</h1>
```
Comment (ericaglimsholt) [inline code/src/pages/CastDetails.js]: This doesn't seems to be changed. But your url is correct and from there you get an id. So when you could use the id here to get the name

Comment (liissee) [reply code/src/pages/CastDetails.js]: Hej! 
How do you mean? I need to get the credit_Id into the second fetch url but I dont know how to do that..

**Pull request closed without merging**